### ISSUE TITLE
Update remaining extractors to use pyClowder 2

### ISSUE THREAD

**Issue by max-zilla** [terraref/computing-pipeline#213]
In order to perform bulk processing efficiently and safely, and to make extractors more robust and fail-resistant, we want to migrate them to use the new PyClowder2 python library instead of the original PyClowder, which has several 'hacks' to support our pipeline that 2 makes formal and complete.

This is not terribly difficult, but I am testing as I go along to make sure behavior does not change.

**repos to update**
- [x] meterological/datparser
- [x] metadata/sensorposition
- [x] metadata/netcdf
- [x] environmental/environmentlogger
- [x] multispectral/flir
- [x] stereo-rgb/demosaic
- [x] stereo-rgb/canopycover
- [x] 3dscanner/ply2las
- [x] hyperspectral/hyperspectral
- [x] lemnatec-indoor/plantcv

**testing with clowder-dev**
deploy updated repo code, install pyClowder 2 on extractor host (including enum and pyyaml if not pip installed yet)
- [x] meterological/datparser
- [x] metadata/sensorposition
- [x] metadata/netcdf
- [x] environmental/environmentlogger
- [x] multispectral/flir
- [x] stereo-rgb/demosaic
- [x] stereo-rgb/canopycover
- [x] 3dscanner/ply2las
- [x] lemnatec-indoor/plantcv

Once all are migrated, our tutorial materials and templates will refer to PyClowder 2 and the corresponding improvements in code structure and ease of use.

**Comment by max-zilla** [comment]
@jterstriep I just created a pull request for my 3D scanner extractor repo.
https://github.com/terraref/extractors-3dscanner/pull/1/files

- attempts to incorporate your changes from the other branch (command line args, setup.py)
- does not move Dockerfile or setup.py outside the ply2las file. The idea is that this repo will eventually hold multiple 3dscanner extractors, ply2las is just one of them. So I think we want to keep separate dockerfiles and setup installers for each.
- I didn't rename terra.ply2las.py to terra_ply2las.py - was that causing problems?
- config.py goes away

I intended for this to be the best of both codebases here. Not going to merge quite yet.

**Comment by jterstriep** [comment]
Part of the configuration is designed around code sharing (eliminating duplication), ease of development, manual and automated testing, automatic builds, and continuous deployments. To meet the needs of all those systems, setup.py and requirements.txt definitely need to be at the root. Of course, this assumes the additional extractors are closely related. If not, then the extractors should be in their own repository.

The Dockerfile is another matter. It might be better to have different directories for each extractor with its own Dockerfile. We might even want a base Dockerfile that builds the common elements and then customize the extractor Dockerfile which might only consist of overriding the CMD line. Alternatively, we could have a general Dockerfile that builds a common image (i.e. a 3dscanner image) that allows any of the extractors to be run from the command line. Either way, I think it's better not to use a custom filename like Docker.<whatever>.  
 
The problem with terra.ply2las.py is that when I try to import from it the extra dot causes confusion. My solution doesn't change the name for users and fixes the import problem. Is there any reason we can't change the filename?

Are we transitioning to pyClowder2 at this point?

**Comment by max-zilla** [comment]
@jterstriep understood, i'll move setup and requirements to the root. The repo extractors will theoretically all operate on the same type of sensor data so it makes sense to keep them together in that regard, even if their purposes may be different.

Will rename the file as well, just couldn't remember what the reason was.

We're transitioning to pyClowder 2, yeah. Primary reason is that @robkooper and I cleaned up a lot of the rough edges of pyC 1 that could cause issues in HPC environments. One big new feature is an error queue - if the extractor tries and fails to process large numbers of messages, we can route them into an error queue in RMQ for examination and resubmit them to processing queue later without losing any information. This was an important new addition before we crunched all these older datasets.
